File: SDKs/amazonaws.com/gamelift/2015-10-01/go/pkg/models/operations/startfleetactions.go
package operations

import (
	"openapi/pkg/models/shared"
)

type StartFleetActionsXAmzTargetEnum string

const (
	StartFleetActionsXAmzTargetEnumGameLiftStartFleetActions StartFleetActionsXAmzTargetEnum = "GameLift.StartFleetActions"
)

type StartFleetActionsHeaders struct {
	XAmzAlgorithm     *string                         `header:"style=simple,explode=false,name=X-Amz-Algorithm"`
	XAmzContentSha256 *string                         `header:"style=simple,explode=false,name=X-Amz-Content-Sha256"`
	XAmzCredential    *string                         `header:"style=simple,explode=false,name=X-Amz-Credential"`
	XAmzDate          *string                         `header:"style=simple,explode=false,name=X-Amz-Date"`
	XAmzSecurityToken *string                         `header:"style=simple,explode=false,name=X-Amz-Security-Token"`
	XAmzSignature     *string                         `header:"style=simple,explode=false,name=X-Amz-Signature"`
	XAmzSignedHeaders *string                         `header:"style=simple,explode=false,name=X-Amz-SignedHeaders"`
	XAmzTarget        StartFleetActionsXAmzTargetEnum `header:"style=simple,explode=false,name=X-Amz-Target"`
}

type StartFleetActionsRequest struct {
	Headers StartFleetActionsHeaders
	Request shared.StartFleetActionsInput `request:"mediaType=application/json"`
}

type StartFleetActionsResponse struct {
	ContentType              string
	InternalServiceException *interface{}
	InvalidRequestException  *interface{}
	NotFoundException        *interface{}
	StartFleetActionsOutput  *shared.StartFleetActionsOutput
	StatusCode               int64
	UnauthorizedException    *interface{}
}

File: SDKs/amazonaws.com/gamelift/2015-10-01/typescript/src/sdk/models/operations/startfleetactions.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";


export enum StartFleetActionsXAmzTargetEnum {
    GameLiftStartFleetActions = "GameLift.StartFleetActions"
}


export class StartFleetActionsHeaders extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Algorithm" })
  xAmzAlgorithm?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Content-Sha256" })
  xAmzContentSha256?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Credential" })
  xAmzCredential?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Date" })
  xAmzDate?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Security-Token" })
  xAmzSecurityToken?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Signature" })
  xAmzSignature?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-SignedHeaders" })
  xAmzSignedHeaders?: string;

  @SpeakeasyMetadata({ data: "header, style=simple;explode=false;name=X-Amz-Target" })
  xAmzTarget: StartFleetActionsXAmzTargetEnum;
}


export class StartFleetActionsRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  headers: StartFleetActionsHeaders;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request: shared.StartFleetActionsInput;
}


export class StartFleetActionsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  internalServiceException?: any;

  @SpeakeasyMetadata()
  invalidRequestException?: any;

  @SpeakeasyMetadata()
  notFoundException?: any;

  @SpeakeasyMetadata()
  startFleetActionsOutput?: shared.StartFleetActionsOutput;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  unauthorizedException?: any;
}
